include/swift/AST/PropertyWrappers.h

```diff
@@ -170,16 +170,38 @@ void simple_display(llvm::raw_ostream &os, PropertyWrapperLValueness l);
 /// be initialized out-of-line using an expression of the wrapped property type.
 PropertyWrapperValuePlaceholderExpr *findWrappedValuePlaceholder(Expr *init);
 
-/// Describes the backing property of a property that has an attached wrapper.
-struct PropertyWrapperBackingPropertyInfo {
+/// The synthesized auxiliary declarations for a wrapped property, including the
+/// backing property wrapper, the projected value variable, and if the wrapped
+/// declaration is a parameter, the local wrapped value variable.
+struct PropertyWrapperAuxiliaryVariables {
   /// The backing property.
   VarDecl *backingVar = nullptr;
 
   /// The synthesized projection property, if any. When present, this takes the name
   /// of the original wrapped property prefixed with \c $
   VarDecl *projectionVar = nullptr;
 
-private:
+  /// The synthesized local wrapped value property, which shadows the original wrapped
+  /// declaration if it is a parameter.
+  VarDecl *localWrappedValueVar = nullptr;
+
+  PropertyWrapperAuxiliaryVariables() {}
+
+  PropertyWrapperAuxiliaryVariables(VarDecl *backingVar, VarDecl *projectionVar,
+                                    VarDecl *localWrappedValueVar = nullptr)
+      : backingVar(backingVar), projectionVar(projectionVar),
+        localWrappedValueVar(localWrappedValueVar) {}
+
+  /// Whether this is a valid property wrapper.
+  bool isValid() const {
+    return backingVar != nullptr;
+  }
+
+  explicit operator bool() const { return isValid(); }
+};
+
+/// Describes the backing property of a property that has an attached wrapper.
```

Comment:
Whoops, this documentation needs to be updated too

Reply:
I forgot to add the comment for that :)

Reply:
✅